src/Tag/Base/BaseNormalTag.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Html\Tag\Base;
+
+use Yiisoft\Html\Html;
+
+/**
+ * Base for all normal tags. Normal tags are the ones that have both opening tag and closing tag.
+ */
+abstract class BaseNormalTag extends Tag
+{
+    private bool $encode = true;
+    private bool $doubleEncode = true;
+    private string $content = '';
+
+    /**
+     * Do not HTML-encode tag content.
+     * @return static
+     */
+    final public function withoutEncode(): self
```

Comment:
I have an idea of naming all these `without*` as `no*` for this package. It's not consistent with another immutable methods but is shorter and is more natural. What do you think?

Reply:
I think this is normal. Also we don't use prefix `with` - and and it looks organic here.

Reply:
I think this ok:

```
withoutEncode → noEncode
withoutDoubleEncode → noDoubleEncode
withoutLabelEncode → noLabelEncode
```

I doubt it in this case:

```
withoutContainer → noContainer
withoutSeparator → noSeparator
```

Reply:
All methods `withoutEncode()` refactored to `encode(bool $encode)`.
Method `withoutSeparator` removed.
Leave only `withoutContainer`. I think this good.